src/data/fin/basic.lean

```diff
@@ -96,6 +96,9 @@ localized "attribute [instance] fact.pow.pos" in fin_fact
 namespace fin
 variables {n m : ℕ} {a b : fin n}
 
+/-- Elimination principle for the empty set `fin 0`, non-dependent version. -/
+def elim {β : Sort*} (x : fin 0) : β := x.elim0
```

Comment:
Are we sure we don't want to call this `elim0'` or similar, as this is still about `fin 0`?

Reply:
Do we ever need the dependent version? Does `λ x, fin.elim x` prove `fin.elim0`?

Reply:
@urkud Yes, it does. But both versions turn out to be useful.

Reply:
@eric-wieser I think this is the most reasonable eliminator for `fin`. So I wouldn't mind if it has the shortest name.

Reply:
Then I think @eric-wieser 's suggestion is to name this one `elim0` and the other `elim0'`, the idea being that `fin.elim` seems too general a name for something about only `fin 0`. I agree, although I don't feel strongly.

Reply:
I agree that this should be named `fin.elim0`. One may assume that `fin.elim` is something like `Π (k : fin n) (f : Π k (hk : k < n), β ⟨k, hk⟩), β k`.